File: TransCoder_tokenized_test_set_functions/python/POSSIBLE_MOVES_KNIGHT.py
def findPossibleMoves ( mat, p, q ) :
    global n, m;
    
    X = [ 2, 1, - 1, - 2, - 2, - 1, 1, 2 ];
    
    Y = [ 1, 2, 2, 1, - 1, - 2, - 2, - 1 ];
    
    count = 0;
    
    for i in range ( 8 ) :
        x = p + X [ i ];
        
        y = q + Y [ i ];
        
        if ( x >= 0 and y >= 0 and x < n and y < m and mat [ x ] [ y ] == 0 ) :
            count += 1;
            
    return count;

File: TransCoder_tokenized_test_set_functions/cpp/POSSIBLE_MOVES_KNIGHT.cpp
int findPossibleMoves ( int mat [ n ] [ m ], int p, int q ) {
  int X [ 8 ] = {
    2, 1, - 1, - 2, - 2, - 1, 1, 2 };
    int Y [ 8 ] = {
      1, 2, 2, 1, - 1, - 2, - 2, - 1 };
      int count = 0;
      for ( int i = 0;
      i < 8;
      i ++ ) {
        int x = p + X [ i ];
        int y = q + Y [ i ];
        if ( x >= 0 && y >= 0 && x < n && y < m && mat [ x ] [ y ] == 0 ) count ++;
      }
      return count;
    }